java-checks/src/main/java/org/sonar/java/checks/MissingDeprecatedCheck.java

```diff
@@ -48,12 +47,7 @@ public void setContext(JavaFileScannerContext context) {
 
   @Override
   public void visitNode(Tree tree) {
-    boolean isLocalVar = false;
-    if (tree.is(Tree.Kind.VARIABLE)) {
-      isLocalVar = currentParent.peek().is(METHOD_KINDS);
-    } else {
-      currentParent.push(tree);
-    }
+    boolean isLocalVar = tree.is(Tree.Kind.VARIABLE) && ((VariableTree) tree).symbol().owner().isMethodSymbol();
```

Comment:
This check seems to be quite useful. Maybe a good candidate to move it in some utils?